### PR TITLE
feat: Deep connector Jira (Metadata)

### LINKED ISSUE #709: Add Jira deep connector
- GetPostAuthInfo
- ListObjectMetadata
- Read
- Write

### PR DESCRIPTION
**Pull request by Cobalt0s**
Closes #709 

# Description

Implemented ListObjectMetadata by making an API call to `field` endpoint which returns fields of Jira issue.
Reference: https://developer.atlassian.com/cloud/jira/platform/rest/v2/api-group-issue-fields/#api-rest-api-2-field-get

# Testing

image

### CODE REVIEW

**Comment by Cobalt0s** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/amp-labs/connectors/799?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#799** <a href="https://app.graphite.dev/github/pr/amp-labs/connectors/799?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#710** <a href="https://app.graphite.dev/github/pr/amp-labs/connectors/710?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 1 other dependent PR ([#755](https://github.com/amp-labs/connectors/pull/755) <a href="https://app.graphite.dev/github/pr/amp-labs/connectors/755?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* **#756** <a href="https://app.graphite.dev/github/pr/amp-labs/connectors/756?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#725** <a href="https://app.graphite.dev/github/pr/amp-labs/connectors/725?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#720** <a href="https://app.graphite.dev/github/pr/amp-labs/connectors/720?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 1 other dependent PR ([#661](https://github.com/amp-labs/connectors/pull/661) <a href="https://app.graphite.dev/github/pr/amp-labs/connectors/661?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* **#712** <a href="https://app.graphite.dev/github/pr/amp-labs/connectors/712?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 1 other dependent PR ([#713](https://github.com/amp-labs/connectors/pull/713) <a href="https://app.graphite.dev/github/pr/amp-labs/connectors/713?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* **#711** <a href="https://app.graphite.dev/github/pr/amp-labs/connectors/711?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @Cobalt0s and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: atlassian/metadata.go [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+	fields, err := c.parseFieldsJiraIssue(rsp.Body)
+	if err != nil {
+		return nil, errors.Join(ErrParsingMetadata, err)
+	}
+
+	// Read response is flattened exposing only important fields which happen to not have id.
+	// To mitigate this API response the Read method will attach id.
+	// Therefore, metadata must include it too.
+	fields["id"] = "Identifier"
```
Comment (jlimatampersand) [inline atlassian/metadata.go:33]:
```suggestion
	fields["id"] = "Id"
```

Id is probably cleaner and is generally used without spelling it out when displaying.

**Review by jlimatampersand — APPROVED** [review]
Overall looks good. But please make sure to return error on this method when "cloudId" is not provided.

File: atlassian/metadata_test.go [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+			t.Parallel()
+
+			defer tt.server.Close()
+
+			connector, err := NewConnector(
+				WithAuthenticatedClient(http.DefaultClient),
+				WithWorkspace("test-workspace"),
+				WithModule(ModuleJira),
+				WithMetadata(map[string]string{
+					"cloudID": "ebc887b2-7e61-4059-ab35-71f15cc16e12", // it doesn't matter for the test
```
Comment (jlimatampersand) [inline atlassian/metadata_test.go:135]: Please change comment since cloudId is now required.

**Pull request merged**